### PR TITLE
[10.0] [MIG] product_supplier_code_purchase

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Product Supplier Code Purchase
==============================

This module adds to the purchase order line the supplier code defined in the product.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by AaronHForgeFlow** [comment]
It's not ADD it's MIG it's merged in v9 https://github.com/OCA/purchase-workflow/pull/376

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test + code review

**Review by LoisRForgeFlow — APPROVED** [review]
Code review, LGTM

File: product_supplier_code_purchase/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                supplier_info = product_supplierinfo_obj.search([
+                    '|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
+                    ('product_id', '=', product.id),
+                    ('name', '=', partner.id)], limit=1)
+                if supplier_info:
+                    code = supplier_info.product_code or ''
+                    line.product_supplier_code = code
+        return True
+
+    product_supplier_code = fields.Char(
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py:28]: Please put fields declaration at the beginning. https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models

File: product_supplier_code_purchase/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    def _compute_product_supplier_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self:
+            partner = line.order_id.partner_id
+            product = line.product_id
+            if product and partner:
+                supplier_info = product_supplierinfo_obj.search([
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py:19]: IMHO you shouldn't do that. On big purchases, your search requests is linearly proportional to your lines amount.

Please use a depends : 
```python
@api.depends('product_id.seller_ids.name', 'product_id.seller_ids.product_code')
```
ORM would do its job and load all you need in self.

Then iterate on self and get the right supplierinfo with a 'filtered':
```python
supplierinfo = line.seller_ids.filtered(lambda s, l=line: s.name == line.partner_id)
```

File: product_supplier_code_purchase/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                    '|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
+                    ('product_id', '=', product.id),
+                    ('name', '=', partner.id)], limit=1)
+                if supplier_info:
+                    code = supplier_info.product_code or ''
+                    line.product_supplier_code = code
+        return True
```
Comment (rousseldenis) [inline product_supplier_code_purchase/models/purchase_order.py:26]: Not needed in compute

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some comments

**Comment by MiquelRForgeFlow** [comment]
Should be closed. Superseeded by https://github.com/OCA/purchase-workflow/pull/535.

**Pull request closed without merging**